pontoon/base/models.py

```diff
@@ -284,53 +284,82 @@ def latest_activity(self, locale=None):
 
         return translations.latest_activity()
 
+    def add_detail(self, array, path, entity, translated, approved):
+        return array.append({
+            'resource__path': path,
+            'resource__entity_count': entity,
+            'translated_count': translated,
+            'approved_count': approved
+        })
+
     @property
     def locales_parts_stats(self):
         """Get project locales with their pages/paths and stats."""
-        pages = self.subpage_set.all()
-        resources = self.resource_set.filter(entity__obsolete=False).distinct()
         details = {}
 
-        for locale in self.locales.all():
-            stats = Stats.objects.filter(resource__in=resources, locale=locale)
+        # Project Locales with prefetched Stats and Resources
+        qs = (
+            Stats.objects
+            .select_related('resource')
+            .filter(resource__project=self, resource__entity__obsolete=False)
+            .distinct()
+        )
+        locales = self.locales.prefetch_related(
+            Prefetch(
+                'stats_set',
+                queryset=qs,
+                to_attr='fetched_stats'
```

Comment:
Is there a reason to use the custom attribute instead of just using `stats_set`?